var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -138,10 +138,15 @@ class Llvm(CMakePackage, CudaPackage):
         description="Link LLVM tools against the LLVM shared library",
     )
     variant(
-        "all_targets",
-        default=False,
-        description="Build all supported targets, default targets "
-        "<current arch>,NVPTX,AMDGPU,CppBackend",
+        "targets",
+        description=("What targets to build. 'auto' corresponds to 'all' in LLVM. "
+                     "Spack's target family is always added (e.g. X86 is automatically "
+                     "enabled when targeting znver2."),
+        values=auto_or_any_combination_of(
+            "AArch64", "AMDGPU", "ARM", "AVR", "BPF", "CppBackend", "Hexagon", "Lanai",
+            "Mips", "MSP430", "NVPTX", "PowerPC", "RISCV", "Sparc", "SystemZ",
+            "WebAssembly", "X86", "XCore"
```

Comment:
Aren't variant values usually lowercase with underscores? It could be confusing when we have a spack arch `aarch64` but have to remember that llvm's usage is `AArch64`. Maybe we need to add standardized variant *values* to https://github.com/spack/spack/pull/25841 as well. But for this particular case, I would try to use Spack arch names  (e.g. `ppc`) and map those to the LLVM naming (`PowerPC`).

Reply:
There's no 1-to-1 mapping between spack targets and llvm targets though, so I'm not sure if we should spack'ify llvm arguments.

Reply:
As a compromise, perhaps we can make it case insensitive for known targets?  I would always default to writing all upper or all lower, have tripped over that on some of the targets more than once.

Reply:
It's lowercase now